helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/ConstraintBasedAlgorithm.java

```diff
@@ -100,7 +100,7 @@ public OptimalAssignment calculate(ClusterModel clusterModel) throws HelixRebala
           getNodeWithHighestPoints(replica, nodes, clusterModel.getContext(), busyInstances,
               optimalAssignment);
       // stop immediately if any replica cannot find best assignable node
-      if (maybeBestNode.isEmpty() || optimalAssignment.hasAnyFailure()) {
+      if (!maybeBestNode.isPresent() || optimalAssignment.hasAnyFailure()) {
```

Comment:
any particular reason to change this particular check? this is optional property.

Reply:
thanks.. i thought the data structure is the same, but here method is the issue. thanks.